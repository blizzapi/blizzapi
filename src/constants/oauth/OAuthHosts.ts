/**
 * List of all available OAuth token URIs
 * via https://develop.battle.net/documentation/guides/using-oauth
 */

import {
  RegionIdProperties,
  Uri,
} from '../../types';

const oAuthHosts: RegionIdProperties<Uri> = {
  1: 'https://us.battle.net',
  2: 'https://eu.battle.net',
  3: 'https://apac.battle.net',
  5: 'https://www.battlenet.com.cn',
};

export default Object.freeze(oAuthHosts);
