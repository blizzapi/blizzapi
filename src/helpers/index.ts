export * from "./bnet";
export * from "./fetch";
export * from "./oauth";
export * from "./validators";
