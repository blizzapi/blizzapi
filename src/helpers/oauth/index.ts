export { getAccessTokenObject } from "./getAccessTokenObject";
export { validateAccessToken } from "./validateAccessToken";
