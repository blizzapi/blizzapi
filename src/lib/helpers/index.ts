export default {
  query: (endpoint: string) => {
    console.log(`query: ${endpoint}`);
  }
}