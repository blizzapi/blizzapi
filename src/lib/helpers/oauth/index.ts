export { default as fetchAccessToken } from './fetchAccessToken';
export { default as getAccessToken } from './getAccessToken';
export { default as validateAccessToken } from './validateAccessToken';
