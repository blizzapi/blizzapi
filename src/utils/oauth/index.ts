export * from "./authorizeUris";
export * from "./checkTokenUris";
export * from "./tokenUris";
