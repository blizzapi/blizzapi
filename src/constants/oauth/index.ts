import { oAuthHosts } from "./OAuthHosts";
import { RegionIdProperties, RegionId, Uri } from "../../types";

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> =>
  Object.freeze({
    [RegionId.us]: `${oAuthHosts[RegionId.us]}${endpoint}`,
    [RegionId.eu]: `${oAuthHosts[RegionId.eu]}${endpoint}`,
    [RegionId.kr]: `${oAuthHosts[RegionId.kr]}${endpoint}`,
    [RegionId.cn]: `${oAuthHosts[RegionId.cn]}${endpoint}`,
  });

const authorizeEndpoint = "/oauth/authorize";
const tokenEndpoint = "/oauth/token";
const checkTokenEndpoint = "/oauth/check_token?token=";

export const authorizeUris = getRegionPropertyArray(authorizeEndpoint);
export const tokenUris = getRegionPropertyArray(tokenEndpoint);
export const checkTokenUris = getRegionPropertyArray(checkTokenEndpoint);
