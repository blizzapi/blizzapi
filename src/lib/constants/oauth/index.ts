import { RegionIdProperties } from '../../../../@interfaces';
import { Uri } from '../../../../@types';
import OAuthHosts from './OAuthHosts';

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> => ({
  1: `${OAuthHosts[1]}${endpoint}`,
  2: `${OAuthHosts[2]}${endpoint}`,
  3: `${OAuthHosts[3]}${endpoint}`,
  5: `${OAuthHosts[5]}${endpoint}`,
});

const authorizeEndpoint = '/oauth/authorize';
const tokenEndpoint = '/oauth/token';
const checkTokenEndpoint = '/oauth/check_token?token=';

const authorizeUris = getRegionPropertyArray(authorizeEndpoint);
const tokenUris = getRegionPropertyArray(tokenEndpoint);
const checkTokenUris = getRegionPropertyArray(checkTokenEndpoint);

export default {
  authorizeUris: Object.freeze(authorizeUris),
  tokenUris: Object.freeze(tokenUris),
  checkTokenUris: Object.freeze(checkTokenUris),
}
