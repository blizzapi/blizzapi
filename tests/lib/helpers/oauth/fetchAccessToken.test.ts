import { fetchAccessToken } from '../../../../src/lib/helpers/oauth';

describe('fetchAccessToken()', () => {
  const oauthUri = 'http://sample.oauth.uri/';
  const clientId = 'sample client id';
  const clientSecret = 'sample client secret';

  test('should be defined', () => {
    expect(fetchAccessToken).toBeDefined();
  });

  test('should be function', () => {
    expect(fetchAccessToken).toBeTruthy();
    expect(typeof fetchAccessToken).toBe('function');
  });

  test('returns correct response for GET request to a valid uri', async () => {
    const response = await fetchAccessToken({ oauthUri, clientId, clientSecret });
    expect(response).toMatchSnapshot();
  });
});