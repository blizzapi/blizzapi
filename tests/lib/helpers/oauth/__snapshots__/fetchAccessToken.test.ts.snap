// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchAccessToken() returns correct response for GET request to a valid uri 1`] = `
Object {
  "access_token": "c2FtcGxlX2NsaWVudF9pZDpzYW1wbGVfY2xpZW50X3NlY3JldA==",
  "headers": Object {
    "Authorization": "Basic c2FtcGxlX2NsaWVudF9pZDpzYW1wbGVfY2xpZW50X3NlY3JldA==",
  },
  "title": "sample mocked data for POST request to http://sample.oauth.uri/",
}
`;
